lldb/test/API/lang/swift/variables/move_function_async/TestSwiftMoveFunctionAsync.py

```diff
@@ -0,0 +1,171 @@
+# TestSwiftMoveFunctionAsync.py
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ------------------------------------------------------------------------------
+"""
+Check that we properly show variables at various points of the CFG while
+stepping with the move function.
+"""
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+import os
+import sys
+import unittest2
+
+def stderr_print(line):
+    sys.stderr.write(line + "\n")
+
+class TestSwiftMoveFunctionAsyncType(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    @swiftTest
+    def test_swift_move_function_async(self):
+        """Check that we properly show variables at various points of the CFG while
+        stepping with the move function.
+        """
+        self.build()
+
+        self.exec_artifact = self.getBuildArtifact(self.exec_name)
+        self.target = self.dbg.CreateTarget(self.exec_artifact)
+        self.assertTrue(self.target, VALID_TARGET)
+
+        self.do_setup_breakpoints()
+
+        self.process = self.target.LaunchSimple(None, None, os.getcwd())
+        threads = lldbutil.get_threads_stopped_at_breakpoint(
+            self.process, self.breakpoints[0])
+        self.assertEqual(len(threads), 1)
+        self.thread = threads[0]
+
+        self.do_check_copyable_value_test()
+        self.do_check_copyable_var_test()
+
+    def setUp(self):
+        TestBase.setUp(self)
+        self.main_source = "main.swift"
+        self.main_source_spec = lldb.SBFileSpec(self.main_source)
+        self.exec_name = "a.out"
+
+    def add_breakpoints(self, name, num_breakpoints):
+        pattern = 'Set breakpoint {} here {}'
+        for i in range(num_breakpoints):
+            pat = pattern.format(name, i+1)
+            brk = self.target.BreakpointCreateBySourceRegex(
+                pat, self.main_source_spec)
+            self.assertGreater(brk.GetNumLocations(), 0, VALID_BREAKPOINT)
+            yield brk
+
+    def get_var(self, name):
+        frame = self.thread.frames[0]
+        self.assertTrue(frame.IsValid(), "Couldn't get a frame.")
```

Comment:
@adrian-prantl do you want me to run to source breakpoint instead of continuing?

Reply:
If you use run_to_source_breakpoint it will set all breakpoints matching the regex and stop at the first one. If it doesn't raise an exception it's safe assume that frame is valid.

Reply:
I actually can't do that in this PR for this test. It is b/c there is a stepping issue where we step backwards due to an instruction having a SILLocation that is out of order. The information around what values are alive is different there and so the test fails. If possible, I would like to get this in to test out the actual functionality and then loop back around and figure out the stepping issue.